### ISSUE TITLE
Code coverage 

### ISSUE THREAD

**Issue by listepo** [balderdashy/sails#2833]
I propose to add code coverage for sails (https://coveralls.io/r/balderdashy/sails) and other sails components.

**Comment by tjwebb** [comment]
I agree! We've just set this up for waterline. @dmarcelino @devinivy interested in applying your code coverage awesomeness to sails?

**Comment by listepo** [comment]
Ping

**Comment by devinivy** [comment]
Should be fairly straightforward just getting the badge!
See: https://github.com/balderdashy/waterline/pull/956/files

**Comment by dmarcelino** [comment]
The badge is the easy part, the more difficult part is setting up the coverage analysis itself (https://github.com/balderdashy/waterline/pull/906/files). I'll have to checkout sails code and take a look, maybe more towards the end of the week.

**Comment by listepo** [comment]
It would be good if the added code coverage, would simplify the writing test, and implement dependency service pack now offers new features

**Comment by sailsbot** [comment]
Thanks for posting, @listepo.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
